core/src/main/java/org/mskcc/cbio/portal/scripts/ImportTabDelimData.java

```diff
@@ -460,46 +466,64 @@ else if (!entrez.matches("[0-9]+")) {
                         }
                     }
 
-                    if (genes == null || genes.isEmpty()) {
-                        genes = Collections.emptyList();
+                    //  If targetLine is specified and does not match the current line, skip the current line.
+                    if (targetLine != null && !(parts[0].equals(targetLine))) {
+                        return false;
                     }
 
-                    //  If no target line is specified or we match the target, process.
-                    if (targetLine == null || parts[0].equals(targetLine)) {
-                        if (genes.isEmpty()) {
-                            //  if gene is null, we might be dealing with a micro RNA ID
-                            if (geneSymbol != null && geneSymbol.toLowerCase().contains("-mir-")) {
-//                                if (microRnaIdSet.contains(geneId)) {
-//                                    storeMicroRnaAlterations(values, daoMicroRnaAlteration, geneId);
-//                                    numRecordsStored++;
-//                                } else {
-                                    ProgressMonitor.logWarning("microRNA is not known to me:  [" + geneSymbol
-                                        + "]. Ignoring it "
-                                        + "and all tab-delimited data associated with it!");
-                                    return false;
-//                                }
+                    List<CanonicalGene> genesMatchingAnAlias = Collections.emptyList();
+                    if (geneSymbol != null) {
+                        genesMatchingAnAlias = daoGene.getGenesForAlias(geneSymbol);
+                    }
+
+                    Set<CanonicalGene> microRNAGenes = new HashSet<>();
+                    Set<CanonicalGene> nonMicroRNAGenes = new HashSet<>();
+                    Iterator<CanonicalGene> geneIterator = Stream.concat(genes.stream(), genesMatchingAnAlias.stream()).iterator();
+                    while (geneIterator.hasNext()) {
+                        CanonicalGene g = geneIterator.next();
+                        if ("miRNA".equals(g.getType())) {
+                            microRNAGenes.add(g);
+                        } else {
+                            nonMicroRNAGenes.add(g);
+                        }
+                    }
+                    if (!microRNAGenes.isEmpty()) {
+                        // for micro rna, duplicate the data
+                        for (CanonicalGene gene : microRNAGenes) {
+                            boolean result = storeGeneticAlterations(values, daoGeneticAlteration, gene, geneSymbol);
```

Comment:
Instead of `boolean result = ` we could do `recordStored = ` and get rid of the next few lines.

Reply:
I think this may not be equivalent. If we dropped the "if (result == true)" line, then recordStored would match the results of the final attempted record. As it is, recordStored winds up true if *any* of the microRNAGenes are imported successfully.

Reply:
Ah yes of course you are right!